flink-kubernetes/src/test/java/org/apache/flink/kubernetes/KubernetesResourceManagerTest.java

```diff
@@ -0,0 +1,372 @@
+			previewPodName,
+			new ArrayList<>(),
+			1024,
+			1,
+			new HashMap<>()));
+		final KubernetesClient client = getKubeClient();
+		assertEquals(1, client.pods().list().getItems().size());
+
+		// Call initialize method to recover worker nodes from previous attempt.
+		resourceManager.initialize();
+
+		final String taskManagerPrefix = CLUSTER_ID + "-taskmanager-";
+		// Register the previous taskmanager, no new pod should be created
+		registerTaskExecutor(new ResourceID(previewPodName));
+		registerSlotRequest(resourceManager);
+		assertEquals(1, client.pods().list().getItems().size());
+
+		// Register a new slot request, a new taskmanger pod will be created with attempt2
+		registerSlotRequest(resourceManager);
+		assertEquals(2, client.pods().list().getItems().size());
+		assertThat(client.pods().list().getItems().stream()
+				.map(e -> e.getMetadata().getName())
+				.collect(Collectors.toList()),
+			Matchers.containsInAnyOrder(taskManagerPrefix + "1-1", taskManagerPrefix + "2-1"));
+
+		resourceManager.getRpcService().stopService().get();
+	}
+
+	private TestingKubernetesResourceManager createAndStartResourceManager(Configuration configuration) throws Exception {
+
+		final TestingRpcService rpcService = new TestingRpcService(configuration);
+		MockResourceManagerRuntimeServices rmServices = new MockResourceManagerRuntimeServices(rpcService, TIMEOUT);
+
+		final TestingKubernetesResourceManager kubernetesResourceManager = new TestingKubernetesResourceManager(
+			rpcService,
+			"kubernetesResourceManager",
+			ResourceID.generate(),
+			configuration,
+			rmServices.highAvailabilityServices,
+			rmServices.heartbeatServices,
+			rmServices.slotManager,
+			rmServices.jobLeaderIdService,
+			new ClusterInformation("localhost", 1234),
+			testingFatalErrorHandler,
+			UnregisteredMetricGroups.createUnregisteredResourceManagerMetricGroup()
+		);
+		kubernetesResourceManager.start();
+		rmServices.grantLeadership();
+		return kubernetesResourceManager;
+	}
+
+	private void registerSlotRequest(TestingKubernetesResourceManager resourceManager) throws Exception {
+
+		CompletableFuture<?> registerSlotRequestFuture = resourceManager.runInMainThread(() -> {
+			resourceManager.getSlotManager().registerSlotRequest(
+				new SlotRequest(new JobID(), new AllocationID(), ResourceProfile.UNKNOWN, jobManagerHost));
+			return null;
+		});
+		registerSlotRequestFuture.get();
+	}
+
+	private void registerTaskExecutor(ResourceID resourceID) throws Exception {
+		final TaskExecutorGateway taskExecutorGateway = new TestingTaskExecutorGatewayBuilder()
+			.setAddress(UUID.randomUUID().toString())
+			.createTestingTaskExecutorGateway();
+		((TestingRpcService) resourceManager.getRpcService()).registerGateway(resourceID.toString(), taskExecutorGateway);
+
+		final ResourceManagerGateway rmGateway = resourceManager.getSelfGateway(ResourceManagerGateway.class);
+
+		final ResourceProfile resourceProfile = ResourceProfile.newBuilder()
```

Comment:
Make it simple to use `ResourceProfile.ZERO` instead?

Reply:
Make sense.